bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -170,6 +172,22 @@ impl Connection {
         Ok(self)
     }
 
+    pub(crate) fn config(&self) -> Option<Config> {
+        let mut raw = core::ptr::null_mut();
+        let config = unsafe {
+            s2n_connection_get_config(self.connection.as_ptr(), &mut raw)
+                .into_result()
+                .ok()
+                .and_then(|_ok| NonNull::new(raw))
+                .map(|raw| Config::from_raw(raw))
+        };
```

Comment:
I would write this a bit different:

```suggestion
        let config = unsafe {
            let mut raw = core::ptr::null_mut();
            s2n_connection_get_config(self.connection.as_ptr(), &mut raw)
                .into_result()
                .ok()?;
            let raw = NonNull::new(raw)?;
            Config::from_raw(raw)
        };
```

Reply:
I completely forgot that since I switched this method's return from Result to Option, I can use "?" to return None! :D Yes that's much better.